pennylane/devices/qubit/measure.py

```diff
@@ -195,13 +196,30 @@ def get_measurement_function(
 
             backprop_mode = math.get_interface(state, *measurementprocess.obs.data) != "numpy"
             if isinstance(measurementprocess.obs, (Hamiltonian, LinearCombination)):
-                # need to work out thresholds for when its faster to use "backprop mode" measurements
-                return sum_of_terms_method if backprop_mode else csr_dot_products
+
+                # need to work out thresholds for when it's faster to use "backprop mode"
+                if backprop_mode:
+                    return sum_of_terms_method
+
+                if not all(obs.has_sparse_matrix for obs in measurementprocess.obs.terms()[1]):
+                    return sum_of_terms_method
+
+                if isinstance(measurementprocess.obs, Hamiltonian) and any(
+                    any(len(o.wires) > 1 for o in qml.operation.Tensor(op).obs)
+                    for op in measurementprocess.obs.ops
+                ):
+                    return sum_of_terms_method
```

Comment:
Why can't we use `csr_dot_products` in this case? Is it because a Hamiltonian with a multi-wire observable has too few zeros to be considered a sparse matrix? If you could provide a clarification comment and/or update the PR description it would be great : )

Reply:
See this line here: https://github.com/PennyLaneAI/pennylane/blob/8d8becea3616d632a581509c2c06a25291d21662/pennylane/ops/qubit/hamiltonian.py#L506

I will add a comment in the code to explain.

Reply:
```suggestion
                # Hamiltonian.sparse_matrix raises a ValueError for this scenario.
                if isinstance(measurementprocess.obs, Hamiltonian) and any(
                    any(len(o.wires) > 1 for o in qml.operation.Tensor(op).obs)
                    for op in measurementprocess.obs.ops
                ):
                    return sum_of_terms_method
```

Reply:
Oh, interesting. Thanks!